src/__tests__/parseWithPointers.spec.ts

```diff
@@ -514,4 +514,126 @@ european-cities: &cities
       expect(result.diagnostics).toEqual([]);
     });
   });
+
+  describe('keys sorting', () => {
+    it('does not retain the order of keys by default', () => {
```

Comment:
Is this a performance thing? Should we retain order of keys by default?

Reply:
> Should we retain order of keys by default?

Not sure, probably not? I believe we want to stay close to `JSON.parse`, don't we?
Either way, this data won't be preserved anyway when the structured copy is performed, so well, Studio won't benefit too much regardless.

Reply:
Does JSON parse not retain key order?

Reply:
> Does JSON parse not retain key order?

It does not.

> If this does not make a meaningful difference on performance, I suggest we do this by default.

The parsing performance is not affected too much. It's around 25-35% slower, which isn't terrible.
The "trapped" access is slower by quite some marging in certain circumstances. I made a few `Object.keys` calls on a few properties from `HugeYaml` fixture, and they took 15-25 times longer.
`Object.getOwnPropertyNames` and `Reflect.ownKeys` aren't particularly slower - only 2-4 times. I didn't test `Object.getOwnPropertySymbols`, since it didn't make any sense.
That said, I guess we can make it disabled by default.